### PR TITLE
refactor(select): Update all public Sass mixins to be called from root

### PR DESCRIPTION
**Pull request by bonniezhou**
Also update `mdc-select--` classes to be added to root element instead of anchor.
Resolves part of issue #4935

### CODE REVIEW

File: packages/mdc-select/README.md [hunk 1]
```diff
@@ -42,9 +42,7 @@ The select uses an [`MDCMenu`](../mdc-menu) component instance to contain the li
 
 > Note: The `data-value` attribute _must_ be present on each option.
 
-The enhanced select requires that you set the `width` of the root element (containing the
-`mdc-select` class) as well as setting the width of the `mdc-select__menu` element to match. This is best done
-through the use of another class (e.g. `demo-width-class` in the example HTML and CSS below).
+The enhanced select requires that you set the `width` of the `mdc-select__anchor` element as well as setting the width of the `mdc-select__menu` element to match. This is best done through the use of another class (e.g. `demo-width-class` in the example HTML and CSS below).
```
Comment (abhiomkar) [inline packages/mdc-select/README.md]: Remove references to _enhanced_ select in this file.

File: packages/mdc-select/_mixins.scss [hunk 2]
```diff
@@ -101,8 +101,7 @@
 
   $masked-radius: mdc-shape-mask-radius($radius, 1 1 0 0);
 
-  &,
```
Comment (abhiomkar) [inline packages/mdc-select/_mixins.scss]: We also need to apply shape to native control to keep the same shape as parent element.

Same below.

File: packages/mdc-select/_mixins.scss [hunk 3]
```diff
@@ -163,18 +162,18 @@
 }
 
 @mixin mdc-select-ink-color_($color) {
-  .mdc-select__native-control,
```
Comment (abhiomkar) [inline packages/mdc-select/_mixins.scss]: Why are these styles removed?

Comment (bonniezhou) [reply packages/mdc-select/_mixins.scss]: The native control class was removed in #4797. @joyzhong to confirm?

File: packages/mdc-select/_mixins.scss [hunk 4]
```diff
@@ -365,16 +356,18 @@
...
 
-  .mdc-select__native-control,
+  .mdc-select__anchor {
+    @include mdc-states-base-color(transparent);
+    @include mdc-floating-label-shake-animation(text-field-outlined);
+    @include mdc-notched-outline-floating-label-float-position($mdc-select-outlined-label-position-y, 0);
+    overflow: visible;
+  }
+
   .mdc-select__selected-text {
```
Comment (abhiomkar) [inline packages/mdc-select/_mixins.scss]: Merge below styles here. Selectors are duplicated.

File: packages/mdc-select/_mixins.scss [hunk 5]
```diff
@@ -365,16 +356,18 @@
...
 
   border: none;
-  overflow: visible;
 
-  .mdc-select__native-control,
+  .mdc-select__anchor {
+    @include mdc-states-base-color(transparent);
+    @include mdc-floating-label-shake-animation(text-field-outlined);
+    @include mdc-notched-outline-floating-label-float-position($mdc-select-outlined-label-position-y, 0);
+    overflow: visible;
```
Comment (abhiomkar) [inline packages/mdc-select/_mixins.scss]: Empty line after `@include` rule.

File: packages/mdc-select/_mixins.scss (hunk 2 above)
Comment (bonniezhou) [reply packages/mdc-select/_mixins.scss]: Native control was removed (see comment below)

**Pull request merged**